### PR TITLE
Add types to array-index package

### PR DESCRIPTION
**Pull request by dimfeld**

### CODE REVIEW

File: packages/array-index/index.tests.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import index = require('./index');
+
+// OK
+index([ { a: 5 }, { a: 6 } ], 'a');
+index([ { a: 5 }, { a: 6 }, null, undefined ], 'a');
+
+// Not OK
+// @ts-expect-error
+index([ { a: 5 }, { a: 6 } ], 'b');
```
Comment (angus-c) [inline packages/array-index/index.tests.ts:9]: this is valid (does not throw an error), just returns `{}`

File: packages/array-index/index.tests.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// @ts-expect-error
+index([ { a: 5 }, { a: 6 } ], (o: number) => o.a);
+// @ts-expect-error
+index([1, 2, 3], 'a');
```
Comment (angus-c) [inline packages/array-index/index.tests.ts:13]: this is valid (does not throw an error), just returns {}

File: packages/array-index/index.tests.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+// OK
+index([ { a: 5 }, { a: 6 } ], 'a');
+index([ { a: 5 }, { a: 6 }, null, undefined ], 'a');
+
+// Not OK
+// @ts-expect-error
+index([ { a: 5 }, { a: 6 } ], 'b');
+// @ts-expect-error
+index([ { a: 5 }, { a: 6 } ], (o: number) => o.a);
```
Comment (angus-c) [inline packages/array-index/index.tests.ts:11]: I'd prefer not to use typescript in the test itself
So just
`index([ { a: 5 }, { a: 6 } ], (o) => o.a);`

**Review by angus-c — CHANGES_REQUESTED** [review]
A few tests things (sorry should have looked for these earlier)

**Comment by dimfeld** [comment]
Thanks for the feedback! Made all three changes.

**Pull request merged**